e2e/create-pvc_test.go

```diff
@@ -67,69 +70,71 @@ var _ = Describe("CreatePVC", func() {
-				Resources: k8sv1.ResourceRequirements{
-					Requests: k8sv1.ResourceList{
-						"storage": quantity,
-					},
-				},
-				StorageClassName: &storageClassName,
-			},
-		}
+		pod := podWithPvcSpec(podName, pvcName, cmd, args)
+		By("creating a pod that attaches pvc")
+		newPod, err := tenantClient.CoreV1().Pods(namespace).Create(context.Background(), pod, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
 
-		pod := &k8sv1.Pod{
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: podName,
-			},
-			Spec: k8sv1.PodSpec{
-				RestartPolicy: k8sv1.RestartPolicyNever,
-				Containers: []k8sv1.Container{
-					{
-						Name:    podName,
-						Image:   image,
-						Command: cmd,
-						Args:    args,
-						VolumeMounts: []k8sv1.VolumeMount{
-							{
-								Name:      volumeName,
-								MountPath: "/opt",
-							},
-						},
-					},
-				},
-				Volumes: []k8sv1.Volume{
-					{
-						Name: volumeName,
-						VolumeSource: k8sv1.VolumeSource{
-							PersistentVolumeClaim: &k8sv1.PersistentVolumeClaimVolumeSource{
-								ClaimName: pvc.GetName(),
-							},
-						},
-					},
-				},
-			},
+		By("Wait for pod to reach a running phase")
+		Eventually(func() error {
+			updatedPod, err := tenantClient.CoreV1().Pods(namespace).Get(context.Background(), newPod.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if updatedPod.Status.Phase != k8sv1.PodRunning {
+				return fmt.Errorf("Pod in phase %s, expected Running", updatedPod.Status.Phase)
+			}
+			return nil
+		}, 3*time.Minute, 5*time.Second).Should(Succeed(), "pod should reach running state")
+
+	})
+	//       kubernetes.io/hostname: kvcluster-control-plane-z6jcv
+
+	It("creates a pvc and attaches to pod", Label("pvcCreation"), func() {
+		pvcName := "test-pvc"
+		storageClassName := "kubevirt"
+
+		podName := "test-pod"
+		cmd := []string{"sh"}
+		args := []string{"-c", "while true; do ls -la /opt; echo this file system was made availble using kubevirt-csi-driver; mktmp /opt/test-XXXXXX; sleep 1m; done"}
+
+		nodes, err := tenantClient.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		// select at least two node names
+		if len(nodes.Items) < 2 {
+			Skip("Can only run with 2 or more tenant nodes")
 		}
+		host1 := nodes.Items[0].Labels[hostNameLabelKey]
+		host2 := nodes.Items[1].Labels[hostNameLabelKey]
 
+		pvc := pvcSpec(pvcName, storageClassName, "1Gi")
 		By("creating a pvc")
 		_, err = tenantClient.CoreV1().PersistentVolumeClaims(namespace).Create(context.Background(), pvc, metav1.CreateOptions{})
 		Expect(err).ToNot(HaveOccurred())
 
-		By("creating a pod that attaches pvc")
+		pod := podWithPvcSpec(podName, pvcName, cmd, args)
+		// add toleration so we can use control node for tests
+		pod.Spec.Tolerations = []k8sv1.Toleration{{
+			Key:      "node-role.kubernetes.io/master",
+			Operator: k8sv1.TolerationOpExists,
+			Effect:   k8sv1.TaintEffectNoSchedule,
+		}}
```

Comment:
it's probably fine to simply add this to all pods created by the generic podWithPvcSpec function

Reply:
ok